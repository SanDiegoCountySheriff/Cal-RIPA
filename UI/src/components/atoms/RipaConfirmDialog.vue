<template>
  <v-dialog v-model="showDialog" max-width="350px" persistent>
    <v-card>
      <v-card-title>{{ title }}</v-card-title>
      <v-spacer></v-spacer>
      <v-card-subtitle>{{ subtitle }}</v-card-subtitle>
      <v-card-actions class="tw-mb-4 tw-w-full">
        <v-spacer></v-spacer>
        <div class="tw-flex tw-justify-end tw-w-full">
          <v-btn color="blue darken-1" class="tw-mr-4" @click="handleCancel">
            No
          </v-btn>
          <v-btn
            color="blue darken-1"
            class="tw-w-24"
            @click="handleConfirm"
            :disabled="isConfirmDisabled"
          >
            Yes
          </v-btn>
        </div>
        <v-spacer></v-spacer>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: 'ripa-confirm-dialog',

  data() {
    return {
      isConfirmDisabled: false,
    }
  },

  methods: {
    handleCancel() {
      this.onClose()
    },

    handleConfirm() {
      this.isConfirmDisabled = true
      this.viewModel = false
      this.onConfirm()
      this.handleCancel()
    },
  },

  watch: {
    showDialog(value) {
      this.showDialog = value
      if (value) {
        this.isConfirmDisabled = false
      }
    },
  },

  props: {
    title: {
      type: String,
      default: '',
    },
    subtitle: {
      type: String,
      default: '',
    },
    showDialog: {
      type: Boolean,
      default: false,
    },
    onClose: {
      type: Function,
      required: true,
    },
    onConfirm: {
      type: Function,
      required: true,
    },
  },
}
</script>
