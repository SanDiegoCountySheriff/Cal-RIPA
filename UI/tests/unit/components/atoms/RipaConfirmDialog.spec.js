import RipaConfirmDialog from '@/components/atoms/RipaConfirmDialog.vue'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Confirm Dialog', () => {
  let vuetify
  let wrapper = null

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  beforeAll(() => {
    jest.spyOn(console, 'log').mockImplementation(() => {})
    jest.spyOn(console, 'warn').mockImplementation(() => {})
  })

  const factory = propsData => {
    return shallowMount(RipaConfirmDialog, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  const testData = {
    title: 'Title',
    subtitle: 'Subtitle',
    showDialog: true,
  }

  it('should match snapshot', async () => {
    wrapper = mount(RipaConfirmDialog, { vuetify, propsData: { ...testData } })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should handle cancel', () => {
    wrapper = factory(testData)

    wrapper.vm.handleCancel()

    expect(wrapper.emitted('on-close')).toBeTruthy()
  })

  it('should handle confirm', () => {
    wrapper = factory(testData)

    wrapper.vm.handleConfirm()

    expect(wrapper.emitted('on-confirm')).toBeTruthy()
  })

  it('should watch showDialog', async () => {
    wrapper = factory(testData)

    wrapper.setProps({ showDialog: false })
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.showDialog).toBeFalsy()

    wrapper.setProps({ showDialog: false })
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.showDialog).toBeFalsy()

    wrapper.setProps({ showDialog: true })
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.showDialog).toBeTruthy()
    expect(wrapper.vm.isConfirmDisabled).toBeFalsy()
  })
})
