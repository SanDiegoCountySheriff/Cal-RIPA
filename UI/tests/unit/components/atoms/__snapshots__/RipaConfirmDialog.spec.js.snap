// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Confirm Dialog should match snapshot 1`] = `
<div class="v-dialog__container">
  <div role="dialog" aria-modal="true" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
    <transition-stub name="dialog-transition" appear="true">
      <div tabindex="0" class="v-dialog v-dialog--active v-dialog--persistent" style="transform-origin: center center; max-width: 350px;">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">Title</div>
          <div class="spacer"></div>
          <div class="v-card__subtitle">Subtitle</div>
          <div class="v-card__actions tw-mb-4 tw-w-full">
            <div class="spacer"></div>
            <div class="tw-flex tw-justify-end tw-w-full"><button type="button" class="tw-mr-4 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default blue darken-1"><span class="v-btn__content">
          No
        </span></button> <button type="button" class="tw-w-24 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default blue darken-1"><span class="v-btn__content">
          Yes
        </span></button></div>
            <div class="spacer"></div>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;
